Day X: “Abbreviations done right: The &lt;abbr&gt; element and why not use it” / Alexander Muzenhardt

alexmuzenhardt opened the pull request:
Include the full article content as Markdown, and adjusting the front matter metadata.

alexmuzenhardt committed:
chore(front-matter): update metadata in Markdown files

Adjust the meta information between the '---' sections in the Markdown
front matter to ensure correct and consistent data representation.

alexmuzenhardt committed:
feat(content): add main article body in Markdown format

Include the full article content as Markdown, completing the document
beyond the front matter metadata.

matuzo commented on hell/adventcalendar/2025/33/index.md:
```diff
@@ -1,35 +1,98 @@
 
-Some text.
+Perfect. Exactly what you were looking for. Semantic, accessible — what could go wrong?  
+So you start using `<abbr>` everywhere, only to get immediate feedback telling you what your abbreviations mean *and* an accessibility report complaining about … those very same abbreviations.
```
I don't understand that sentence. The structure looks a little too German. :) Can you rephrase that maybe?

alexmuzenhardt replied:
Sure, I will change the sentence. 👍

alexmuzenhardt replied:
Changed it with commit 18e68498. Please check again. Thanks.

matuzo commented on hell/adventcalendar/2025/33/index.md:
```diff
@@ -1,35 +1,98 @@
 
-<p class="highlight"><strong>Note:</strong> Some text.</p>
+# The problem with abbr
+The biggest problem with the `<abbr>` element is that it behaves completely inconsistently across browsers, and some assistive technologies do not read parts of it at all.  
```
> and some assistive technologies do not read parts of it at all.

That sounds like the element is broken, but really the problem is support for the title attribute. Please rephrase and consider condensing the statement by merging this sentence with the next one.

alexmuzenhardt replied:
I would just change the meaning of the sentence if this is ok. Like this:

"The biggest problem with the `<abbr>` element is that its title attribute behaves completely inconsistently across browsers, and some assistive technologies do not read parts of it at all."

This does not really change the sentence too much but it changes the meaning that the title attribute is not supported well instead of the abbr element and it underpins the following sentence with the tests from Adrian.

Please tell me, if this approach is fine for you. I will change it then.

alexmuzenhardt replied:
Changed it with commit c53f7277. Please check again, thanks.

matuzo reviewed (commented):
Great stuff, thank you!

alexmuzenhardt committed:
Merge branch 'matuzo:advent25' into advent25

alexmuzenhardt committed:
refactor(content): improve sentence readability and structure

Reword a sentence in the article to enhance clarity and flow by adjusting
its phrasing and structure.

alexmuzenhardt committed:
fix(content): correct misleading statement about abbr and title attribute

Adjust a sentence that previously implied the HTML abbr element was the
issue. Clarify that the problem relates to the title attribute used with
abbr, correcting the intended meaning.

matuzo commented:
Great, thank you, @alexmuzenhardt. Are you happy, can I merge it?

alexmuzenhardt commented:
@matuzo I am happy with it. When you think it is fine too, you can merge it.
